### PR TITLE
Updated "Use __dict__ and ignore __slots__ on classes #169

### LINKED ISSUE #168: Using __slots__ hides class variables
If ``__slots__`` is declared in a class, it takes precedence over everything, including class methods and variables. However ``__slots__`` has an effect on instance attributes only, not class attributes.

The relevant lines is https://github.com/astropy/sphinx-automodapi/blob/0aad86aa90bcd7ab5b3004b732e21a3bd3d9de70/sphinx_automodapi/automodsumm.py#LL534C33-L549C33. Maybe ``names`` should also be filled with ``getattr(type(obj), '__dict__').keys()`` or something like that ?

### PR DESCRIPTION
**Pull request by kylefawcett**
Fixes #168 and adds test case to prevent regression

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,8 @@ Changes in sphinx-automodapi
 0.17.0 (unreleased)
 -------------------
 
+- Fixes fixes issue where ``__slots__`` hides class variables [#168]
```
Comment (pllim) [inline CHANGES.rst:7]:
```suggestion
- Fixes fixes issue where ``__slots__`` hides class variables [#181]
```

**Comment by pllim** [comment]
@Nodd , do you want to have a look at the added tests?

Thanks, all!

**Comment by kylefawcett** [comment]
@pllim Circling back around on this. I just cleaned up the slots example module to be more precisely consistent with python nomenclature and added a basic test for the example to meet the coverage thresholds. The other existing examples don't have tests so I'm not sure if you want to set a precedent. I can just pop off the last commit if you don't want tests for the slot example module.

**Comment by pllim** [comment]
More tests is good, so no need to undo it. Thanks!

I tag some people for review because I am not familiar with slots or automodapi. But if no one responds, I can take a stab at reviewing after the holidays.

Happy holidays!

**Comment by kylefawcett** [comment]
Happy New Year! Anyone available to review this?

**Comment by pllim** [comment]
Hi, as people are just returning from their holiday breaks, I would give it another week or two. Thanks for your patience!

**Comment by pllim** [comment]
There are 2 jobs that failed. Can you please rebase and see if they go away? Thanks!

**Comment by pllim** [comment]
If rebase has nothing to do, please let me know, then I can  restart those jobs directly. Thanks!

**Review by bsipocz — APPROVED** [review]
I'm not familiar with __slots__ either, but dug up the history of this part of the code. It was added many years ago in https://github.com/astropy/sphinx-automodapi/pull/53, along with test cases, which still pass after this PR.

So I would say go ahead and merge this.

**Comment by bsipocz** [comment]
Thank you @kylefawcett!

**Pull request merged**